From: uazo <uazo@users.noreply.github.com>
Date: Wed, 15 Apr 2026 14:23:46 +0000
Subject: Enables keyword-based site search on Android devices

Enables keyword-based site search functionality on Android devices

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 .../chromium/chrome/browser/omnibox/UrlBarMediator.java   | 8 +++++---
 .../org/chromium/components/omnibox/OmniboxFeatures.java  | 2 +-
 components/omnibox/common/omnibox_features.cc             | 1 +
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/UrlBarMediator.java
@@ -86,9 +86,11 @@ class UrlBarMediator implements UrlBarTextContextMenuDelegate {
         mModel.set(UrlBarProperties.RICH_TEXT_CHANGE_LISTENER, this::onRichTextChanged);
         mModel.set(UrlBarProperties.KEY_DOWN_LISTENER, keyDownListener);
         mModel.set(UrlBarProperties.SHOW_HINT_TEXT, true);
-        mModel.set(
-                UrlBarProperties.MANAGE_SEARCH_ENGINES_CALLBACK,
-                this::onManageSearchEnginesClicked);
+        // if (OmniboxFeatures.sOmniboxSiteSearch.isEnabled()) {
+        //     mModel.set(
+        //             UrlBarProperties.MANAGE_SEARCH_ENGINES_CALLBACK,
+        //             this::onManageSiteSearchClicked);
+        // }
         setBrandedColorScheme(BrandedColorScheme.APP_DEFAULT);
         pushTextToModel(/* originChanged= */ false);
     }
diff --git a/components/omnibox/common/android/java/src/org/chromium/components/omnibox/OmniboxFeatures.java b/components/omnibox/common/android/java/src/org/chromium/components/omnibox/OmniboxFeatures.java
--- a/components/omnibox/common/android/java/src/org/chromium/components/omnibox/OmniboxFeatures.java
+++ b/components/omnibox/common/android/java/src/org/chromium/components/omnibox/OmniboxFeatures.java
@@ -135,7 +135,7 @@ public class OmniboxFeatures {
             newFlag(OmniboxFeatureList.POST_DELAYED_TASK_FOCUS_TAB, FeatureState.ENABLED_IN_PROD);
 
     public static final CachedFlag sOmniboxSiteSearch =
-            newFlag(OmniboxFeatureList.OMNIBOX_SITE_SEARCH, FeatureState.ENABLED_IN_TEST);
+            newFlag(OmniboxFeatureList.OMNIBOX_SITE_SEARCH, FeatureState.ENABLED_IN_PROD);
 
     public static final CachedFlag sStarterPackExpansion =
             newFlag(OmniboxFeatureList.STARTER_PACK_EXPANSION, FeatureState.ENABLED_IN_PROD);
diff --git a/components/omnibox/common/omnibox_features.cc b/components/omnibox/common/omnibox_features.cc
--- a/components/omnibox/common/omnibox_features.cc
+++ b/components/omnibox/common/omnibox_features.cc
@@ -305,6 +305,7 @@ BASE_FEATURE(kOmniboxSearchPrefetchOnEnterKeyDown, DISABLED);
 
 // Enables keyword-based site search functionality on Android devices.
 BASE_FEATURE(kOmniboxSiteSearch, DISABLED);
+SET_CROMITE_FEATURE_ENABLED(kOmniboxSiteSearch);
 
 // Enables additional site search providers for the Site search Starter Pack.
 BASE_FEATURE(kStarterPackExpansion, enable_if(!IS_IOS));
--
